src/Features/Core/Portable/MetadataAsSource/IMetadataDocumentPersister.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.MetadataAsSource;
+
+/// <summary>
+/// Manages persisting metadata documents (outside of the workspace).
+/// For example, it can persist generated metadata documents to disk or handle them in memory.
+/// </summary>
```

Comment:
i'm actually curious why we ever persist to disk, if we have memory as an option.  It seems  like that would eliminate a bunch of annoyances :)

Reply:
oh.  i think i remember.  is this because in VS it needs the Tab/Buffer to be backed by something on disk?

--

if that's the case, then @olegtk Does VS now allow us to open a buffer where the contents are purely in memory?  Or would that need to be a feature request (effectively, for it to support 'virtual files' like VSCode does)?

Thanks!

Reply:
Yeah, unfortunately VS still requires file on disk, we tried to improve things during Nexus, but only achieved marginal improvements...